### PR TITLE
Refactoring dig commmand

### PR DESCRIPTION
**Pull request by rugwirobaker**
* add roundtrip func﻿

### CODE REVIEW

File: internal/cli/internal/command/command.go [hunk 1]
```diff
@@ -408,21 +408,11 @@ func RequireAppName(ctx context.Context) (context.Context, error) {
...
-	name := flag.GetApp(ctx)
-	if name == "" {
-		// if there's no flag present, first consult with the environment
-		if name = env.First("FLY_APP"); name == "" {
-			// and then with the config file (if any)
-			if cfg := app.ConfigFromContext(ctx); cfg != nil {
-				name = cfg.AppName
-			}
-		}
+	if err != nil && err == errRequireAppName {
```
Comment (azazeal) [inline internal/cli/internal/command/command.go:413]: use `errors.Is` here, please.

File: internal/cli/internal/command/command.go [hunk 2]
```diff
@@ -408,21 +408,11 @@ func RequireAppName(ctx context.Context) (context.Context, error) {
...
+		err = nil
 	}
 
-	return app.WithName(ctx, name), nil
+	return ctx, err
```
Comment (azazeal) [inline internal/cli/internal/command/command.go:417]: I think this may be introducing a bug.

**Review by azazeal — COMMENTED** [review]
a few starting thoughts

File: internal/cli/internal/command/command.go [hunk 3]
```diff
@@ -408,21 +408,15 @@ func RequireAppName(ctx context.Context) (context.Context, error) {
...
 
-	name := flag.GetApp(ctx)
-	if name == "" {
-		// if there's no flag present, first consult with the environment
-		if name = env.First("FLY_APP"); name == "" {
-			// and then with the config file (if any)
-			if cfg := app.ConfigFromContext(ctx); cfg != nil {
-				name = cfg.AppName
-			}
+	if err != nil {
```
Comment (azazeal) [inline internal/cli/internal/command/command.go:413]: `errors.Is` understands nils, you don't need to nest your ifs here.

File: internal/cli/internal/command/command.go [hunk 4]
```diff
@@ -408,21 +408,15 @@ func RequireAppName(ctx context.Context) (context.Context, error) {
...
+		if errors.Is(err, errRequireAppName) {
+			return ctx, nil
```
Comment (azazeal) [inline internal/cli/internal/command/command.go:415]: if you get an error you should set the name to an empty string. here you're returning a context which doesn't carry a name so callers of `app.NameFromContext` will crash.

File: internal/cli/internal/command/dig/dig.go [hunk 5]
```diff
@@ -238,10 +194,54 @@ func run(ctx context.Context) error {
...
 }
 
-// ResolverForOrg takes a connection to the wireguard agent and an organization
+// roundTrip a DNS request across a "TCP" socket; we'd just use miekg/dns's Client, but I don't think it promises to
+// work over our weird UDS TCP proxy.
+func roundTrip(conn net.Conn, m *dns.Msg) (*dns.Msg, error) {
+	m.Id = dns.Id()
+	m.Compress = true
+
+	m.Id = dns.Id()
```
Comment (azazeal) [inline internal/cli/internal/command/dig/dig.go:203]: why you set the ID and compress twice?

**Review by azazeal — CHANGES_REQUESTED** [review]
this can be cleaned up a bit further

File: internal/cli/internal/command/dig/dig.go (hunk 5 above)
Comment (rugwirobaker) [reply internal/cli/internal/command/dig/dig.go:203]: Copy pasted code🤦🏽‍♂️

File: internal/cli/internal/command/command.go (hunk 4 above)
Comment (rugwirobaker) [reply internal/cli/internal/command/command.go:415]: You are right I should have tested that path

File: internal/cli/internal/command/dig/dig.go [hunk 6]
```diff
@@ -238,10 +194,51 @@ func run(ctx context.Context) error {
...
-// ResolverForOrg takes a connection to the wireguard agent and an organization
+// roundTrip a DNS request across a "TCP" socket; we'd just use miekg/dns's Client, but I don't think it promises to
+// work over our weird UDS TCP proxy.
+func roundTrip(conn net.Conn, m *dns.Msg) (*dns.Msg, error) {
+	m.Id = dns.Id()
+	m.Compress = true
+
+	buf, err := m.Pack()
+	if err != nil {
+		return nil, fmt.Errorf("dns round trip: %w", err)
```
Comment (azazeal) [inline internal/cli/internal/command/dig/dig.go:205]: this isn't a dns round trip error; I wouldn't decorate my errors in a helper function like `roundTrip` but at the caller.

**Review by azazeal — CHANGES_REQUESTED** [review]

File: internal/cli/internal/command/dig/dig.go (hunk 6 above)
Comment (rugwirobaker) [reply internal/cli/internal/command/dig/dig.go:205]: @azazeal  Yeah indeed. But if i's, not a round trip error, what decoration would give it in the main function?

File: internal/cli/internal/command/dig/dig.go [hunk 7]
```diff
@@ -226,7 +182,7 @@ func run(ctx context.Context) error {
 	case "TXT-NATIVE":
 		txts, err := r.LookupTXT(ctx, name)
 		if err != nil {
-			return FixNameError(err, ns)
+			return fixNameError(err, ns)
 		}
 
 		fmt.Printf("%s\n", strings.Join(txts, ""))
```
Comment (azazeal) [inline internal/cli/internal/command/dig/dig.go:188]: please use `iostreams`

File: internal/cli/internal/command/dig/dig.go (hunk 6 above)
Comment (azazeal) [reply internal/cli/internal/command/dig/dig.go:205]: just don't decorate them and bubble them up. all of these are errors due to writing to the connection - it'll be obvious where they're coming from

**Review by azazeal — CHANGES_REQUESTED** [review]

**Review by azazeal — APPROVED** [review]

**Pull request merged**